### ISSUE TITLE
Can Records be mutable?

### ISSUE THREAD

**Issue by MNF** [dotnet/docs#21726]
“Records are immutable by default.” 
I interpreted this sentence as it’s also possible to make them mutable. But from what I’ve read next, I understand that Records are ALWAYS immutable.
If the records can be mutable, it will be good to mention, how/when they can be changed.
Otherwise could you please remove “ by default” and leave “Records are immutable”
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c36588d0-6bdf-1740-220c-10b20bb88730
* Version Independent ID: 587ed47f-f532-e018-9b0d-6ee76afd20e4
* Content: [What's new in C# 9.0 - C# Guide](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9#record-types)
* Content Source: [docs/csharp/whats-new/csharp-9.md](https://github.com/dotnet/docs/blob/master/docs/csharp/whats-new/csharp-9.md)
* Product: **dotnet-csharp**
* Technology: **csharp-whats-new**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by owenneil** [comment]
Records can definitely be mutable. The short form of records like this are immutable:
`public record MyPositionalRecord(string ImmutableValue)`

But if you use the longer form, you can change the mutability by using non-init setters. For example:
```csharp
public record MyRecord
{
    public string MyMutableProperty { get; set; }
    public string MyImmutableProperty { get; init; }
}
```
Then used:
```csharp
var rec = new MyRecord
{
    MyMutableProperty = "InitialValue",
    MyImmutableProperty = "Immutable"
}

rec.MyMutableProperty = "NewValue"
```

**Comment by Youssef1313** [comment]
There are two types of immutability:

1. Shallow immutability:


    Only simple positional records of the form `record R(some params...);` enforces shallow immutability.

2. Deep immutability:

    There is no way to enforce deep immutability, even with simple positional records.

A record in general can always have some property with a setter, which will break shallow immutability.

I find the sentence you quoted from the docs confusing.

**Comment by BillWagner** [comment]
The question of immutability in records has a lot of nuances. Either form *can* be immutable, or mutable. Consider these variations of the example above:

```csharp
public record MyPositionalRecord(string ImmutableValue);
public record MyImmutableRecord
{
    public string ImmutableValue { get; init; };
}

// mutable variations:
public record MyMutablePositionalRecord(string ImmutableValue)
{
    public string MyMutableProperty { get; set; }
}
public record MyRecord
{
    public string MyMutableProperty { get; set; }
    public string MyImmutableProperty { get; init; }
}
```

It gets more complicated if you add inheritance:

```csharp
public record MyPositionalRecord(string ImmutableValue);
public record MyDerivedMutableRecord(string ImmutableValue) : MyPositionalRecord(ImmutableValue)
{
    public string MyMutableProperty { get; set; }
}
```

The key points are:

1. All `record` types follow value semantics for equality.
1. None of the synthesized members will modify the state of a record.
1. Any properties declared as positional parameters are init-only properties.

Does that clarify the explanation?

**Comment by BillWagner** [comment]
This was fixed in a PR #23699
